spec/acceptance/init_spec.rb

```diff
@@ -1,74 +1,75 @@
 # run a test task
 require 'spec_helper_acceptance'
+require 'pry'
 
 describe 'package task' do
-  include Beaker::TaskHelper::Inventory
-  include BoltSpec::Run
-
   operating_system_fact = os[:family]
   redhat_six = os[:family] == 'redhat' && os[:release].to_i == 6
 
-  def bolt_config
-    { 'modulepath' => RSpec.configuration.module_path }
+  before(:all) do
+    inventory_hash = inventory_hash_from_inventory_file
+    inventory_hash = add_feature_to_group(inventory_hash, 'puppet-agent', 'ssh_nodes')
+    inventory_hash = add_feature_to_group(inventory_hash, 'puppet-agent', 'winrm_nodes')
+    inventory_hash = add_feature_to_group(inventory_hash, 'puppet-agent', 'docker_nodes')
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
```

Comment:
Perhaps my local testing is wrong but I don't think this is actually modifying the `puppet-agent` feature for the targets being executed on. For example, the inventoryfile looks something like:
```
---
groups:
- name: docker_nodes
  nodes:
  - name: 2913d12e181cd850587a9795e42485e9fc56816392b1c2e26759876e11c376c6
    config:
      transport: docker
      docker:
        shell-command: bash -lc
    facts:
      provisioner: docker_exp
      container_id: 2913d12e181cd850587a9795e42485e9fc56816392b1c2e26759876e11c376c6
      platform: waffleimage/debian8
  - name: 8f0c27136bf35629976ced3ccbe979864ae81bb1d4b8087adcb458553378deab
    config:
      transport: docker
      docker:
        shell-command: bash -lc
    facts:
      provisioner: docker_exp
      container_id: 8f0c27136bf35629976ced3ccbe979864ae81bb1d4b8087adcb458553378deab
      platform: waffleimage/debian9
  - name: 6f66e3ef7a7806273677551d57bc572359fe4679cb005d616bb969dfdfaf3bfa
    config:
      transport: docker
      docker:
        shell-command: bash -lc
    facts:
      provisioner: docker_exp
      container_id: 6f66e3ef7a7806273677551d57bc572359fe4679cb005d616bb969dfdfaf3bfa
      platform: waffleimage/ubuntu14.04
  - name: 02c57f5fafd85d8180a5328283e318659767bb00f918102c23b57d621656b90c
    config:
      transport: docker
      docker:
        shell-command: bash -lc
    facts:
      provisioner: docker_exp
      container_id: 02c57f5fafd85d8180a5328283e318659767bb00f918102c23b57d621656b90c
      platform: waffleimage/ubuntu16.04
  - name: b2237aacb0753470faa2b84bbcad62f4091f762b71aa291ba8cc925bb26c5bc4
    config:
      transport: docker
      docker:
        shell-command: bash -lc
    facts:
      provisioner: docker_exp
      container_id: b2237aacb0753470faa2b84bbcad62f4091f762b71aa291ba8cc925bb26c5bc4
      platform: waffleimage/ubuntu18.04
  features: []
- name: ssh_nodes
  nodes: []
  features: []
- name: winrm_nodes
  nodes: []
  features: []
```
So adding and removing features to an empty group does nothing. 

I think that what is happening is that the shell task is actually running for every single target because the puppet-agent feature is never set for any of the groups. Perhaps this is a just my setup, here is what I did
```
 2018  PLATFORMS=debian_puppet5 bundle exec rake 'litmus:provision_list[waffle_deb]'
 2019  PLATFORMS=debian_puppet5 bundle exec rake 'litmus:install_agent[puppet5]'
 2020  PLATFORMS=debian_puppet5 bundle exec rake litmus:install_module
 2021  bundle exec rake litmus:acceptance:parallel
 2022  bundle exec rake --tasks
 2023  PLATFORMS=debian_puppet5 bundle exec rake litmus:acceptance:b2237aacb0753470faa2b84bbcad62f4091f762b71aa291ba8cc925bb26c5bc4
```

Reply:
we have used remove_from_feature_group to remove feature_agent from inventory.yaml file hence we are not able to see them after the test case execution. Was able to add and remove to inventory.yaml file successfully.
But I got another fix to check before adding if it already exists.